open_event/helpers/data.py

```diff
@@ -0,0 +1,179 @@
+"""Copyright 2015 Rafal Kowalski"""
+from sqlalchemy.orm.collections import InstrumentedList
+from flask import flash
+
+from ..models import db
+from ..models.track import Track
+from ..models.session import Session
+from ..models.speaker import Speaker
+from ..models.sponsor import Sponsor
+from ..models.microlocation import Microlocation
+from ..helpers.update_version import VersionUpdater
+
+
+def update_version(event_id, is_created, column_to_increment):
+    VersionUpdater(event_id=event_id,
+                   is_created=is_created,
+                   column_to_increment=column_to_increment).update()
+
+class DataManager(object):
+    @staticmethod
+    def create_track(form, event_id):
+        new_track = Track(name=form.name.data,
+                          description=form.description.data,
+                          event_id=event_id)
+        new_track.session = form.session.data
+        db.session.add(new_track)
+        db.session.commit()
+        update_version(event_id, True, "tracks_ver")
+
+    @staticmethod
+    def update_track(form, track):
+        data = form.data
+        del data['session']
+        db.session.query(Track)\
+            .filter_by(id=track.id)\
+            .update(dict(data))
+        track.session = form.session.data
+        db.session.add(track)
+        db.session.commit()
+        update_version(track.event_id, False,"tracks_ver")
+
+    @staticmethod
+    def remove_track(track_id):
+        track = Track.query.get(track_id)
+        db.session.delete(track)
```

Comment:
Here is a function you can use to reduce repeated code: https://gist.github.com/leto/76541d4b0d34ffeb906b